### ISSUE TITLE
Weights should be "normalizable"

### ISSUE THREAD

**Issue by jzwar** [tataratat/splinepy#299]
We need some function `normalize_weights`, that sets the biggest weight to 1. All multiplications of the weight vector with a scalar do not change the representation of the spline geometry. 

$$
C(x) = \frac{\sum_i w_i C_i B_i(x)}{\sum_j w_j B_j(x)} =  \frac{\sum_i (\alpha w_i) C_i B_i(x)}{\sum_j (\alpha w_j) B_j(x)} 
$$

However, in order to perform compositions between splines, the weighted control points must fit the ND unit cube.

This spline
```python
rational = sp.RationalBezier(
    degrees=[2],
    control_points=[[0,0],[0.5,1],[1,0]],
    weights=[1,2,1]
)
```
should work as this on:
```python
rational = sp.RationalBezier(
    degrees=[2],
    control_points=[[0,0],[0.5,1],[1,0]],
    weights=[0.5,1,0.5]
)
```
however, the former raises an exception...

**Comment by j042** [comment]
what kind or exception?

**Comment by jzwar** [comment]
`bezman` checks the `WeightedControlPoints` if they are within bounds `[0,1]`
I could also modify the check in busman, so it checks non-weighted ctps, but I still think that this is something we should provide

**Comment by j042** [comment]
Is it "illegal" to have weights outside of [0, 1]? If not, I think raising exception is a wrong behavior from bezman

**Comment by j042** [comment]
is this solved by #324?

**Comment by jzwar** [comment]
The behavior has been changed in bezman, but weights are not normalizable yet. I still think we can close this issue as you can just multiply with the inverse of the maximum weight
